x-pack/plugins/ml/public/application/datavisualizer/index_based/page.tsx

```diff
@@ -107,7 +108,7 @@ export const Page: FC = () => {
     autoRefreshSelector: true,
   });
 
-  const dataLoader = new DataLoader(currentIndexPattern, kibanaConfig);
+  const dataLoader = new DataLoader(currentIndexPattern, getToastNotifications());
```

Comment:
Please avoid usting cached `getToastNotifications` and retriveve `toastNotifications` from the kibana context.

Reply:
The rest of the file uses the same approach and I didn't plan to refactor it in this PR.